tests/s25Main/UI/testImageButton.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2005 - 2022 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "controls/ctrlImageButton.h"
+#include "ogl/ITexture.h"
+#include "uiHelper/uiHelpers.hpp"
+#include <rttr/test/random.hpp>
+#include <turtle/mock.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace rttr::test;
+
+namespace {
+MOCK_BASE_CLASS(TestTexture, ITexture)
+{
+public:
+    TestTexture(Extent size) : size_(size) {}
+    Position GetOrigin() const override { return Position::all(0); }
+    Extent GetSize() const override { return size_; }
+    MOCK_NON_CONST_METHOD(DrawFull, 2) // LCOV_EXCL_LINE
+    MOCK_NON_CONST_METHOD(Draw, 3)
+
+    Extent size_;
+};
+} // namespace
+
+// LCOV_EXCL_START
+namespace mock {
+static mock::stream& operator<<(mock::stream& s, const Rect& r)
+{
+    auto size = r.getSize();
+    return s << '(' << mock::format(r.left) << ',' << mock::format(r.top) << ',' << mock::format(size.x) << ','
+             << mock::format(size.y) << ')';
+}
+} // namespace mock
+// LCOV_EXCL_STOP
+
+BOOST_FIXTURE_TEST_SUITE(ImageButton, uiHelper::Fixture)
+
+BOOST_AUTO_TEST_CASE(DrawSmallerImageThanButton)
+{
+    TestTexture img(Extent(20, 10));
+    MOCK_EXPECT(img.Draw).once().with(Rect(20, 15, 20, 10), Rect(0, 0, 20, 10), 0xFFFFFFFFu);
+
+    Window wnd(nullptr, randomValue<unsigned>(), DrawPoint::all(0));
+    ctrlImageButton btn(&wnd, randomValue<unsigned>(), DrawPoint(10, 5), Extent(40, 30), TextureColor::Green1, &img,
+                        "");
+    btn.Draw();
+}
+
+BOOST_AUTO_TEST_CASE(DrawLargerImageThanButton)
+{
+    // Note the 2px border to prevent dirtying 3d button's ridge
```

Comment:
That comment should be 1 line below so it is above to what it refers to not above another line
And maybe explain what is meant by "2px" border. I.e. that the image is shifted, drawn with 4px reduced size and the center part of the image is chosen, i.e. `(Extend(100,80) +- Extend(36,26))/ 2`  (the 2nd Rect)

See those behavior test case strategies used by other projects which here would go like `When(DrawLargerImageThanButton).Then(UseCenterPieceOfImage AND Offset by border from all 4 sides)`

Reply:
I'm not sure if I get what you mean, especially in 2nd part, but I've reworded the comment to be more descriptive